state/cleanup.go

```diff
@@ -1668,7 +1669,7 @@ func (st *State) obliterateUnit(ctx context.Context, store objectstore.ObjectSto
 		opErrs = append(opErrs, err)
 	}
 	if err == nil {
-		err = applicationService.DestroyUnit(ctx, unitName)
+		err = applicationService.DestroyUnit(ctx, coreunit.Name(unitName))
```

Comment:
**todo:** If it is safe to not une NewName here (and validate), add a comment that state it.